pandas/tools/plotting.py

```diff
@@ -2274,7 +2276,7 @@ def _plot(data, x=None, y=None, subplots=False,
             plot_obj = klass(data, x=x, y=y, subplots=subplots, ax=ax,
                              kind=kind, **kwds)
         else:
-            raise ValueError('Invalid chart type given %s' % kind)
+            raise ValueError(invalid_ct_fmtstr % kind)
```

Comment:
I think you should ideally differentiate between the two messages. As the first message is just general 'not a valid plot kind'. While this message is specific for DataFrames (you get here if you give a kind that is only possible for a dataframe, but if your object is not a dataframe). So here I would say something like '%r is only a valid plot kind for DataFrames' or the like.